### PR TITLE
Add messages to serial failures

### PR DESCRIPTION
**Pull request by rwx788**
Resurrection of #977 with the fix with proper initialization + unit tests for parse_serial_output_qemu.

[Verification run](http://g226.suse.de/tests/2079#step/welcome/8).

See [poo#36448](https://progress.opensuse.org/issues/36448).

### CODE REVIEW

File: t/17-basetest.t [hunk 1]
```diff
@@ -11,19 +12,70 @@ BEGIN {
...
+        {type => 'soft', message => 'DoNotMatch', pattern => qr/DoNotMatch/},
+    ];
+    $basetest->parse_serial_output_qemu(0);
+    is($basetest->{result}, undef, 'test result untouched without match');
+    is($message,            undef, 'test details do not have extra message');
+
+    $basetest->{serial_failures} = [
+        {type => 'soft', message => 'SimplePattern', pattern => qr/Serial/},
+    ];
+    # push @$expected_details, {title => 'Serial Failure', result => 'softfail', text => ''};
```
Comment (okurz) [inline t/17-basetest.t]: Disabled code?

**Comment by rwx788** [comment]
tinycv compilation fails. @foursixnine could you please give me a hint?

**Comment by Martchus** [comment]
@rwx788 Note that the messages after the tinycv `g++` line are just warnings which are also printed on successful test runs. So tinycv doesn't fail to compile here. The same counts for the message about `Module::Pluggable`. Unfortunately I have no clue what's going wrong so far except that it is something in target `check-local`.

BTW: We can not easily fix the source of the distracting C++ warnings because the relevant code is in a header provided by Perl itself. Maybe just disable them by adding `-Wno-literal-suffix` to compile flags?

**Comment by rwx788** [comment]
@Martchus thanks for the comment, I guess I've found root cause of the failure.

**Comment by Martchus** [comment]
And what was it? (Since you amended your existing commits it is hard to figure from the diff.)

**Comment by rwx788** [comment]
I had quotes for hash keys entry. Error was visible when I executed check-local on my side.

File: basetest.pm [hunk 2]
```diff
@@ -638,34 +638,44 @@ sub search_for_expected_serial_failures {
...
-
-                    $self->record_testresult($fail_type);
-                    $self->record_resultfile('Serial Failure', "Serial error: $line", result => $fail_type);
-                    $self->{result} = $fail_type;
-
+        for my $regexp_table (@{$failures}) {
+            my $regexp  = $regexp_table->{pattern};
+            my $message = $regexp_table->{message};
+            my $type    = $regexp_table->{type};
+            die "Wrong type defined for serial failure. Only soft or hard allowed." if $type !~ /soft|hard/;
```
Comment (foursixnine) [inline basetest.pm]: Please add the type in the message in both cases of the `die`  call

Comment (rwx788) [reply basetest.pm]: Fixed, thanks!

**Pull request merged**